Lib/fnmatch.py

```diff
@@ -77,24 +77,33 @@ def translate(pat):
     There is no way to quote meta-characters.
     """
 
-    STAR = object()
-    parts = _translate(pat, STAR, '.')
-    return _join_translated_parts(parts, STAR)
+    parts, indices = _translate(pat, '*', '.')
+    return _join_translated_parts(parts, indices)
 
+_re_setops_sub = re.compile(r'([&~|])').sub
+_re_escape = functools.lru_cache(maxsize=32768)(re.escape)
 
-def _translate(pat, STAR, QUESTION_MARK):
+def _translate(pat, star, question_mark):
     res = []
     add = res.append
+    indices = []
```

Comment:
To be clear what these indices are:

```suggestion
    star_indices = []
```

(will need other changes below)